Sources/StreamChat/Database/DTOs/ChannelDTO.swift

```diff
@@ -177,6 +177,14 @@ extension NSManagedObjectContext {
         dto.lastMessageAt = payload.lastMessageAt?.bridgeDate
         dto.memberCount = Int64(clamping: payload.memberCount)
         dto.truncatedAt = payload.truncatedAt?.bridgeDate
+        // We don't always set `truncatedAt` directly since we also use this property as `hiddenAt`
+        // when a channel is hidden with `clearHistory`
+        // Backend doesn't send `truncatedAt` for this case and we reset it, causing the hidden messages to re-appear
+        // It's not possible to set `truncatedAt` to `nil` once it's set on backend side
+        // so it's safe to keep client-side changes when backend sends `nil`
+        if payload.truncatedAt != nil {
+            dto.truncatedAt = payload.truncatedAt?.bridgeDate
```

Comment:
This sounds dangerous, and it would be much better to clear the message history, as stated in the Slack discussion. Any reason why you went with this approach?

Reply:
Even if we cleared the message history, we still need this "workaround" as long as we use `truncatedAt` as `hiddenAt`
Alternatively,
```swift
- channelDTO.truncatedAt = event.createdAt.bridgeDate
+ channelDTO.messages = channelDTO.messages.filter({ $0.createdAt.bridgeDate > event.createdAt })
```
but i'm not sure of the performance implication of this. Doable though.

Why does it sound dangerous? There's no way to un-truncate a channel by setting `truncatedAt` to nil, so we're safe to mutate `truncatedAt` locally as long as backend keeps sending `null` for it.

Reply:
Do we have a timescale for the backend implementation? It feels like OK to do this but we should really push for the backend implementation